libavformat/http.c

```diff
@@ -1283,6 +1291,21 @@ static void bprint_escaped_path(AVBPrint *bp, const char *path)
     }
 }
 
+#define unlikely(x) __builtin_expect(!!(x),0)
+
+/* This might be updated by parsing nonceExpireTime from command json */
+int64_t nonce_expire_time = ((int64_t)60 * 60 - 3) * 1000000; /* 1 hour - 3 second by default.
+                                                                 * We're doing that -3 since we don't
+                                                                 * know exact nonce birth time, so we
+                                                                 * start trying to acquire new one 3
+                                                                 * seconds in advance */
+
+static void http_invalidate_auth(HTTPAuthState *s)
+{
+    if (unlikely(av_gettime() - s->used_nonce_birth_time > nonce_expire_time))
```

Comment:
Interesting, I did not know about `__buitin_expect()`. 
Not really a crucial code path to do performance optimizations in as it will only run a couple of times per second.